posthog/views.py

```diff
@@ -58,15 +59,48 @@ def handler(request, *args, **kwargs):
 
 
 def health(request):
-    executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
-    plan = executor.migration_plan(executor.loader.graph.leaf_nodes())
-    status = 503 if plan else 200
-    if status == 503:
+    """
+    Validate that everything this process needs to operate correctly is in place.
+    Returns a dict of checks to boolean status, returning 503 status if any of
+    them is non-True
+    """
+    migrations_uptodate = health_migrations()
+    kafka_connected = health_kafka()
+
+    if migrations_uptodate == False:
+        # NOTE: before adding kafka to this health check, we were just checking
+        # migrations, sending the below sentry error, and returning 503. I've
```

Comment:
we have 3 kinds of migrations (django/postgres, clickhouse, async-migrations) - looks like we only check postgres ones here, maybe clarify the name otherwise it could be a bit misleading.

Reply:
I've instead created separate endpoints for k8s, and clarified that this is specifically postgres migrations that we are checking.